### PR TITLE
remove dependency on `apollo-graphql` and port over functions

### LINKED ISSUE #1134: fix(subgraph): Remove dependency on `apollo-graphql`
`apollo-graphql` is an oddball dependency that exists in the to-be-sunsetted `apollo-tooling` repo. It's the only real cross-dependency of its kind among the federation packages which seems to imply these tools might be better suited within the federation monorepo.

Furthermore, a cross-dependency between the two monorepos (specifically the `apollo-graphql` and the `@apollo/federation` packages) is causing a blocking hiccup which prevents the `apollo-graphql` package from being bumped to the next major (`0.10.x`) due to the current version of `@apollo/federation` requiring `^0.9.x`. We could probably take steps to band-aid this instead but at this point I find it's been a thorn long enough and would rather relocate or remove the dependency altogether.

### PR DESCRIPTION
**Pull request by clenfest**
I put the ported functions primarily in `gateway-js`, but I did duplicate an interface inside of `federation-integration-testsuite-js` as well because I was getting a weird compilation error. I thought about creating a new top level directory for this stuff, but decided against it mainly because it would have been more of a hassle, but I can do it if there are strong feelings.

Fixes #1134

### CODE REVIEW

File: gateway-js/src/index.ts [hunk 1]
```diff
@@ -1271,3 +1271,4 @@ export {
 };
 
 export * from './datasources';
+export * from './schema-helper';
```
Comment (trevor-scheer) [inline gateway-js/src/index.ts:1274]: Do we need to export these at the package level? I'd rather them not become part of the public API.

File: subgraph-js/src/buildSubgraphSchema.ts [hunk 2]
```diff
@@ -9,13 +9,14 @@ import {
...
-  buildSchemaFromSDL,
-  transformSchema,
   GraphQLSchemaModule,
-  modulesFromSDL,
-  addResolversToSchema,
   GraphQLResolverMap,
-} from 'apollo-graphql';
+  addResolversToSchema,
+  modulesFromSDL,
+  transformSchema,
```
Comment (trevor-scheer) [inline subgraph-js/src/buildSubgraphSchema.ts:16]: It seems (at least this fn) is only used by the subgraph package, can it live there instead and not in the gateway? Also, this package doesn't define a dependency on gateway (and I would prefer if it continued not to). Are we able to keep it that way?

File: subgraph-js/tsconfig.json [hunk 3]
```diff
@@ -6,5 +6,7 @@
   },
   "include": ["src/**/*"],
   "exclude": ["**/__tests__"],
-  "references": []
+  "references": [
+    { "path": "../gateway-js" },
```
Comment (trevor-scheer) [inline subgraph-js/tsconfig.json:10]: We would also need to add gateway to subgraph's package.json dependencies, but see comment above (I hope we can avoid that)

**Review by trevor-scheer — COMMENTED** [review]
Just a high level skim first, let me know if I really missed the boat on any of my comments.

File: subgraph-js/src/buildSubgraphSchema.ts (hunk 2 above)
Comment (clenfest) [reply subgraph-js/src/buildSubgraphSchema.ts:16]: I think there are three options, I'm happy to do any of them, just let me know which you prefer.
1. Make a new package for the common code and put it there (or in internals-js)
2. Duplicate the code in both gateway-js and subgraph-js. The only function used by both is `addResolversToSchema` + the interfaces
3. Make gateway-js dependent on subgraphs-js.

Comment (trevor-scheer) [reply subgraph-js/src/buildSubgraphSchema.ts:16]: 👍 for option 2 (not too worried about duplication in the short term, should have a solution for this very soon). Given that, I'd rather avoid adding dependencies on each other unless it really makes sense to.

File: gateway-js/src/schema-helper/buildSchemaFromSDL.ts [hunk 4]
```diff
@@ -1,96 +1,14 @@
...
-        return moduleOrSDL;
-      }
-    });
-  } else {
-    return [{ typeDefs: modulesOrSDL }];
-  }
-}
+import { GraphQLResolverMap } from './resolverMap';
 
 export function addResolversToSchema(
```
Comment (trevor-scheer) [inline gateway-js/src/schema-helper/buildSchemaFromSDL.ts:13]: Looks like we can rename this file to `addResolversToSchema` now

**Review by trevor-scheer — APPROVED** [review]
LGTM!

**Pull request merged**